### PR TITLE
Add E2E tests for the Filter Products By Stock Block

### LINKED ISSUE #6323: Filter Products by Stock Block: Implement E2E tests
Add E2E tests for the Filter Products by Attribute Block. We should check that:
- when the filter is applied only the products that match that filter are rendered (on post/page and a PHP classic template).
- only on the classic PHP template the refresh is triggered.
- after the page refreshes, the URL is what we expect.
- when the filter button is enabled the page refresh is only triggered once the user clicks on it.

### PR DESCRIPTION
**Pull request by gigitux**
This PR adds E2E tests for the Filter Products By Attribute Block. These tests check that:
- when the filter is applied only the products that match that filter are rendered.
- only on the classic PHP template the refresh is triggered.
- after the page refreshes, the URL is what we expect.
- when the filter button is enabled the page refresh is only triggered once the user clicks on it.

For testing this filter, it has been necessary to import an out-of-stock product. For this reason, I updated tests related to the `Mini Block`, since that for the out-of-stock products there isn't the possibility to interact with the Mini Cart block.

Fixes #6323 

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

#### User Facing Testing

Check out this branch

1. Run `npm run test:e2e tests/e2e/specs/shopper/filter-products-by-stock.test.ts`.
2. Be sure that tests don't fail.

* [x] Do not include in the Testing Notes

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

File: tests/e2e/globals.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { Page } from 'puppeteer';
+
+declare global {
+	const page: Page;
+}
```
Comment (tomasztunik) [inline tests/e2e/globals.d.ts]: we don't need this anymore right?

File: tests/e2e/specs/shopper/filter-products-by-stock.test.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	page.waitForSelector( selectors.frontend.productsList + '.is-loading', {
+		hidden: true,
+	} );
+
+const goToShopPage = () =>
+	page.goto( BASE_URL + '/shop', {
+		waitUntil: 'networkidle0',
+	} );
+
+const { selectors } = block;
```
Comment (tomasztunik) [inline tests/e2e/specs/shopper/filter-products-by-stock.test.ts]: Can we move it up? It's the same issue that I mentioned in the final comment. We rely on `selectors` being hoisted in the `waitForAllProductsBlockLoaded` util method.

File: tests/e2e/specs/shopper/mini-cart.test.js [hunk 3]
```diff
@@ -227,8 +227,7 @@ describe( 'Shopper → Mini Cart', () => {
 			}
 
 			// Get a random product to better replicate human behavior.
-			const product =
-				products[ Math.floor( Math.random() * products.length ) ];
+			const product = products[ 1 ];
```
Comment (tomasztunik) [inline tests/e2e/specs/shopper/mini-cart.test.js:240]: should we remove the comment if we don't grab a random product anymore?

**Review by tomasztunik — COMMENTED** [review]
Can we move https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6357/files#diff-f248b582a461ea8862aaff45996500c702d332709b92d6346b0fa7544821ed37R55 higher up cause we rely on selectors being hoisted in the `waitFroAllProductsBlockLoaded` that is couple lines above it - can be confusing. There was another similar issue.

In the [tests/e2e/specs/shopper/mini-cart.test.js](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6357/files#diff-9dc97725b56d59c26af45911780454fe4f353b19fea38adb42e35045cbe2c4eb) file we don't use the selectors object pattern and it really hurts the readability. The usage of nth-child etc. makes me wonder - alright but why nth-child what element is it? - I guess it's not a scope of the PR to rewrite the whole file to use this pattern but maybe worth adding it at least around these new updates?

**Review by tomasztunik — APPROVED** [review]
👌  thanks for the updates and adding selectors - much easier to follow!

**Comment by gigitux** [comment]
I'm merging this PR because the failed tests are unrelated to the changes of this PR

**Pull request merged**